cmd/utils/flags.go

```diff
@@ -1310,12 +1315,85 @@ func SetPlsConfig(ctx *cli.Context, stack *node.Node, cfg *pls.Config) {
+	if ctx.GlobalIsSet(PlasmaRootChainContractFlag.Name) {
+		cfg.RootChainContract = common.HexToAddress(ctx.GlobalString(PlasmaRootChainContractFlag.Name))
 	}
-	cfg.RootChainContract = common.HexToAddress(ctx.GlobalString(PlasmaRootChainContractFlag.Name))
 
-	cfg.Genesis = core.DefaultGenesisBlock(cfg.RootChainContract)
+	switch {
+	case ctx.GlobalBool(TestnetFlag.Name):
+		if !ctx.GlobalIsSet(NetworkIdFlag.Name) {
+			cfg.NetworkId = 3
+		}
+		cfg.Genesis = core.DefaultTestnetGenesisBlock()
+	case ctx.GlobalBool(RinkebyFlag.Name):
+		if !ctx.GlobalIsSet(NetworkIdFlag.Name) {
+			cfg.NetworkId = 4
+		}
+		cfg.Genesis = core.DefaultRinkebyGenesisBlock()
+	case ctx.GlobalBool(DeveloperFlag.Name):
+		if !ctx.GlobalIsSet(NetworkIdFlag.Name) {
+			cfg.NetworkId = 1337
+		}
+		// hard coding for dev mode
+		var (
+			operatorKey, _ = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+			rootChainURL   = "ws://localhost:8546"
+
+			_statesRoot       [32]byte
+			_transactionsRoot [32]byte
+			_receiptsRoot     [32]byte
+
+			development      = true
+			NRELength        = big.NewInt(2)
+			statesRoot       = "0ded2f89db1e11454ba4ba90e31850587943ed4a412f2ddf422bd948eae8b164"
+			transactionsRoot = "56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421"
+			receiptsRoot     = "0000000000000000000000000000000000000000000000000000000000000000"
+		)
+
+		t := time.NewTimer(2 * time.Second)
+		defer t.Stop()
+
+		rootchainBackend, err := ethclient.Dial(rootChainURL)
+		if err != nil {
+			Fatalf("Failed to connect rootchain: %v", err)
+		}
+
+		opt := bind.NewKeyedTransactor(operatorKey)
+		eh, _, _, err := epochhandler.DeployEpochHandler(opt, rootchainBackend)
+		if err != nil {
+			Fatalf("Failed to deploy epoch handler contract")
+		}
+
+		<-t.C
+		copy(_statesRoot[:], statesRoot)
+		copy(_transactionsRoot[:], transactionsRoot)
+		copy(_receiptsRoot[:], receiptsRoot)
+
+		rootchainContract, _, _, err := rootchain.DeployRootChain(opt, rootchainBackend, eh, development, NRELength, _statesRoot, _transactionsRoot, _receiptsRoot)
+		if err != nil {
+			Fatalf("Failed to deploy rootchain contract")
+		}
+		log.Info("Deploy rootchain contract", "address", rootchainContract)
+
+		account, err := ks.ImportECDSA(operatorKey, "")
+		if err != nil {
+			Fatalf("Faild to import operator account: %v", err)
+		}
+		log.Info("Unlocking operator account", "address", account.Address)
+
+		if err = ks.Unlock(account, ""); err != nil {
+			Fatalf("Failed to unlock operator account: %v", err)
+		}
+
+		cfg.Genesis = core.DeveloperGenesisBlock(uint64(ctx.GlobalInt(DeveloperPeriodFlag.Name)), rootchainContract, account.Address)
+		cfg.Operator = account
+		cfg.RootChainURL = rootChainURL
+		cfg.RootChainContract = rootchainContract
+
+		if !ctx.GlobalIsSet(MinerGasPriceFlag.Name) && !ctx.GlobalIsSet(MinerLegacyGasPriceFlag.Name) {
+			cfg.MinerGasPrice = big.NewInt(1)
+		}
```

Comment:
Is there any specific reason change `miner.gasprice` option for dev?

Reply:
it is original the code implemented in [go-ethereum](https://github.com/ethereum/go-ethereum/blob/616cf782036de06e8f9795e4c8782363c7de9898/cmd/utils/flags.go#L1382-L1384)